### ISSUE TITLE
 NullInjectorError: No provider for CompilerFactory!

### ISSUE THREAD

**Issue by Indrajith-Sync** [angular/angular#37215]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [X] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

Not sure about the angular version which it is working

### Description

Facing this script error after the project is served using **npm start**

## 🔥 Exception or Error
<pre><code>
scripts.js:5067 Uncaught NullInjectorError: StaticInjectorError(Platform: core)[CompilerFactory]: 
  NullInjectorError: No provider for CompilerFactory!
    at NullInjector.get (http://localhost:4200/scripts.js:5067:29)
    at resolveToken (http://localhost:4200/scripts.js:16262:28)
    at tryResolveToken (http://localhost:4200/scripts.js:16206:20)
    at StaticInjector.get (http://localhost:4200/scripts.js:16100:24)
    at compileNgModuleFactory__PRE_R3__ (http://localhost:4200/scripts.js:32539:40)
    at PlatformRef.bootstrapModule (http://localhost:4200/scripts.js:32781:20)
    at Module../src/main.ts (http://localhost:4200/scripts.js:143265:77)
    at __webpack_require__ (http://localhost:4200/scripts.js:54751:30)
    at Object.0 (http://localhost:4200/scripts.js:143278:18)
    at __webpack_require__ (http://localhost:4200/scripts.js:54751:30)
</code></pre>

## 🌍 Your Environment
![image](https://user-images.githubusercontent.com/29561170/82418831-1fa90380-9a9b-11ea-887f-f7db55300503.png)

## My package.json file

<pre><code>
{
  "name": "custom-element-test",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "postinstall": "ngcc --properties es2015 browser module main --first-only --create-ivy-entry-points",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e",
    "build:richtext-editor": "ng build --extra-webpack-config webpack.externals.js --project richtext-editor  --output-hashing none --outputPath=projects/testapp/src/assets/custom-comps/richtext-editor --single-bundle true"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~9.1.4",
    "@angular/common": "~9.1.4",
    "@angular/compiler": "~9.1.4",
    "@angular/core": "~9.1.4",
    "@angular/forms": "~9.1.4",
    "@angular/platform-browser": "~9.1.4",
    "@angular/platform-browser-dynamic": "~9.1.4",
    "@angular/router": "~9.1.4",
    "@angular/elements": "^9.1.4",
    "@webcomponents/custom-elements": "^1.4.1",
    "@webcomponents/webcomponentsjs": "^2.4.3",
    "rxjs": "~6.5.4",
    "tslib": "^1.10.0",
    "zone.js": "~0.10.2"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.901.4",
    "@angular-devkit/build-ng-packagr": "~0.901.4",
    "@angular/cli": "~9.1.4",
    "@angular/compiler-cli": "~9.1.4",
    "@angular/language-service": "~9.1.4",
    "@types/node": "^12.11.1",
    "@types/jasmine": "~3.5.0",
    "@types/jasminewd2": "~2.0.3",
    "codelyzer": "^5.1.2",
    "jasmine-core": "~3.5.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.3.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~2.1.0",
    "karma-jasmine": "~2.0.1",
    "karma-jasmine-html-reporter": "^1.4.2",
    "protractor": "~5.4.3",
    "ts-node": "~8.3.0",
    "tslint": "~5.18.0",
    "typescript": "~3.7.5",
    "ngx-build-plus": "^9.0.6"
  }
}

</code></pre>

**Comment by pkozlowski-opensource** [comment]
> Facing this script error after the project is served using npm start

I'm sorry but there is not enough details here for us to investigate. We would need a minimal reproduction scenario (as requested in the issue template) to be able to understand and confirm the issue.

**Comment by Indrajith-Sync** [comment]
Hi @pkozlowski-opensource 

Try the following steps in the below sample to reproduce the reported issue.

- npm install
- npm run build:richtext-editor
- npm start

Sample: 
[Angular_issue_Sample.zip](https://github.com/angular/angular/files/4655487/Angular_issue_Sample.zip)

**Comment by pkozlowski-opensource** [comment]
@Indrajith-Sync ohhh, do you think it would be possible to reproduce in https://stackblitz.com/ instead of sharing a zip file? It would speed up investigations immensely...

**Comment by Indrajith-Sync** [comment]
@pkozlowski-opensource  the issue is not reproduced with stackblitz. Hence shared the angular project 

https://stackblitz.com/edit/angular-elements-tutorial-dad5wm?file=package.json

**Comment by JoostK** [comment]
A couple of observations:

1. The stacktrace shows `compileNgModuleFactory__PRE_R3__ ` in `scripts.js` in an Ivy app, this indicates that the Angular sources have not been compiled for usage with Ivy.
2. The project uses `ngx-build-plus` and extended Webpack configuration. I'm not familiar with this setup, but it looks like `ngx-build-plus` creates the `scripts.js` file that contains the non-Ivy bundle of Angular.

ngcc is invoked using `--create-ivy-entry-points` which will generate the Ivy variant of Angular into separate folders, which requires custom module resolution logic. In the default CLI setup this is working by default, but it looks like `ngx-build-plus` may not work like that, or there may be something else at play. As I'm not familiar with the custom builder, the only recommendation I can make is to turn off `--create-ivy-entry-points`, but there may be alternative possibilities.

In conclusion, I don't think this is a framework issue, rather an invalid build setup.

**Comment by alxhub** [comment]
I agree with @JoostK's analysis here - this appears to be an issue with `ngx-build-plus` and how it interacts with Ivy builds in the CLI. Based on that, I'm going to go ahead and close this issue as not actionable for us.

If that's not the case, please feel free open a new issue with a reproduction that doesn't use `ngx-build-plus`.

**Comment by saranglakare** [comment]
Just FYI: I had the same issue today after updating to Angular `9.1.11`. I resolution was to run `npm install` once. Although I did not see it installing any new packages, it did the trick!

**Comment by sistla001** [comment]
I have same issue. Upgraded angular elements project from 7 to 9.1.11. and trying to use it in a spa which is also in 9.1.11. Seeing this exact error. This worked fine when both SPA and elements were in Angular 7.

**Comment by parliament718** [comment]
I'm also having this exact error and stacktrace with Angular 10 and Im not using ngx-build-plus. I'm trying to use @ngtools/webpack in my webpack config.
